projects/plugins/jetpack/modules/search/instant-search/lib/tracks.js

```diff
@@ -1,26 +1,67 @@
+let isAnalyticsEnabled = true;
 const globalProperties = {};
 
+/**
+ * Disable Analytics.
+ */
+export function disableAnalytics() {
+	isAnalyticsEnabled = false;
+}
+
+/**
+ * Initalizes Tracks.
+ */
 export function initializeTracks() {
-	window._tkq = window._tkq || [];
+	if ( isAnalyticsEnabled ) {
+		window._tkq = window._tkq || [];
+	}
 }
+
+/**
+ * Resets current user's tracked identity.
+ */
 export function resetTrackingCookies() {
-	window._tkq.push( [ 'clearIdentity' ] );
+	isAnalyticsEnabled && window._tkq.push( [ 'clearIdentity' ] );
 }
 
+/**
+ * Associates the current site with events fired in the future.
+ *
+ * @param {string} siteId - Current site identifier.
+ */
 export function identifySite( siteId ) {
-	globalProperties.blog_id = siteId;
+	if ( isAnalyticsEnabled ) {
+		globalProperties.blog_id = siteId;
+	}
 }
 
+/**
+ * Fires a general event to Tracks.
+ *
+ * @param {string} eventName - Name of the event.
+ * @param {object} properties - Event properties.
+ */
 export function recordEvent( eventName, properties ) {
-	window._tkq.push( [ 'recordEvent', eventName, { ...globalProperties, ...properties } ] );
+	isAnalyticsEnabled &&
+		window._tkq.push( [ 'recordEvent', eventName, { ...globalProperties, ...properties } ] );
 }
 
+/**
+ * Fires a TrainTracks render event to Tracks.
+ *
+ * @param {object} properties - Event properties.
+ */
 export function recordTrainTracksRender( properties ) {
-	recordEvent( 'jetpack_instant_search_traintracks_render', properties );
+	isAnalyticsEnabled && recordEvent( 'jetpack_instant_search_traintracks_render', properties );
```

Comment:
```suggestion
	recordEvent( 'jetpack_instant_search_traintracks_render', properties );
```
`recordEvent` already tested whether analytics is enabled or not.

Reply:
👍